src/elements/content-picker/stories/tests/ContentPicker-visual.stories.js

```diff
@@ -17,8 +25,708 @@ export const selectedEmptyState = {
         const selectedButton = canvas.getByRole('button', { name: '0 Selected' });
         await userEvent.click(selectedButton);
         await waitFor(() => {
-            expect(canvas.getByText('You haven’t selected any items yet.')).toBeInTheDocument();
+            expect(canvas.getByText("You haven't selected any items yet.")).toBeInTheDocument();
+        });
+    },
+    parameters: {
+        msw: {
+            handlers: [
+                http.get(`${DEFAULT_HOSTNAME_API}/2.0/folders/69083462919`, () => {
+                    return HttpResponse.json(mockRootFolder);
+                }),
+            ],
+        },
+    },
+};
+
+export const emptyFolder = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        await waitFor(
+            () => {
+                // Verify empty folder state using partial match
+                expect(canvas.getByText(/no items/i)).toBeInTheDocument();
+            },
+            {
+                timeout: SLEEP_TIMEOUT * 3, // Increase timeout to allow for retries
+            },
+        );
+    },
+    parameters: {
+        msw: {
+            handlers: [
+                http.get(`${DEFAULT_HOSTNAME_API}/2.0/folders/69083462919`, () => {
+                    return HttpResponse.json(mockEmptyRootFolder);
+                }),
+            ],
+        },
+    },
+};
+
+export const emptySelectionMode = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+        const chooseButton = canvas.getByLabelText('Choose');
+        await userEvent.click(chooseButton);
+        expect(chooseButton).toBeDisabled();
+    },
+};
+
+export const withError = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        await waitFor(
+            () => {
+                // Verify error state is displayed - check for exact error message
+                expect(canvas.getByText('A network error has occurred while trying to load.')).toBeInTheDocument();
+            },
+            {
+                timeout: SLEEP_TIMEOUT * 3, // Increase timeout to allow for retries
+            },
+        );
+    },
+    parameters: {
+        msw: {
+            handlers: [
+                http.get(`${DEFAULT_HOSTNAME_API}/2.0/folders/69083462919`, () => {
+                    return HttpResponse.error();
+                }),
```

Comment:
if you return a 500 instead you dont need the timeout on the line above because it will not retry